### PR TITLE
Add software AES encryption to RFM95 transport HAL

### PR DESCRIPTION
**Pull request by fallberg**
Also clean up capabilities flags a bit and fix test-sketch
for rfm95 to use proper encryption flag.

### CODE REVIEW

**Comment by fallberg** [comment]
This needs testing by someone with a RFM95 rig. I don't have one unfortunately.

File: hal/transport/RFM95/MyTransportRFM95.cpp [hunk 1]
```diff
@@ -48,11 +70,26 @@ uint8_t transportGetAddress(void)
...
+#if defined(MY_RFM95_ENABLE_ENCRYPTION)
+	// copy input data because it is read-only
+	(void)memcpy(RFM95_dataenc,data,len);
+	// has to be adjusted, WIP!
+	RFM95_aes.set_IV(0);
+	const uint8_t finalLength = len > 16 ? 32 : 16;
+	//encrypt data
+	RFM95_aes.cbc_encrypt(RFM95_dataenc, RFM95_dataenc, finalLength /16);
+	if (noACK) {
+		(void)RFM95_sendWithRetry(to, RFM95_dataenc, len, 0, 0);
```
Comment (tekka007) [inline hal/transport/RFM95/MyTransportRFM95.cpp]: finalLength required here (due to block size)

File: hal/transport/RFM95/MyTransportRFM95.cpp [hunk 2]
```diff
@@ -48,11 +70,26 @@ uint8_t transportGetAddress(void)
...
+		return true;
+	}
+	return RFM95_sendWithRetry(to, RFM95_dataenc, len);
```
Comment (tekka007) [inline hal/transport/RFM95/MyTransportRFM95.cpp]: same here, finalLength

File: hal/transport/RFM95/MyTransportRFM95.cpp (hunk 1 above)
Comment (fallberg) [reply hal/transport/RFM95/MyTransportRFM95.cpp]: Thanks

File: core/MyTransport.cpp [hunk 3]
```diff
@@ -647,7 +647,7 @@ void transportProcessMessage(void)
 	const uint8_t msgLength = min(mGetLength(_msg), (uint8_t)MAX_PAYLOAD);
 	// calculate expected length
 	const uint8_t expectedMessageLength = HEADER_SIZE + (mGetSigned(_msg) ? MAX_PAYLOAD : msgLength);
-#if defined(MY_RF24_ENABLE_ENCRYPTION) || defined(MY_NRF5_ESB_ENABLE_ENCRYPTION) || defined(MY_RFM95_ENABLE_ENCRYPTION)
+#if defined(MY_ENCRYPTION_FEATURE)
```
[the remark this answers is not in the document]
Comment (fallberg) [reply core/MyTransport.cpp]: Ah, I see.

File: hal/transport/RFM95/MyTransportRFM95.cpp (hunk 2 above)
Comment (fallberg) [reply hal/transport/RFM95/MyTransportRFM95.cpp]: Pretty sure I had this at some point, but a rebase might have nuked it. Will fix.

**Comment by fallberg** [comment]
Requested changes done

**Comment by fallberg** [comment]
I suggest we go ahead with this but mention that the feature is untested in the release notes.

**Comment by tekka007** [comment]
Tested RFM95 AES encryption with simple password option and E2P-based AES key on GW and node.

**Pull request merged**